### PR TITLE
Initial Match and Network filters example

### PR DESCRIPTION
**Pull request by nmartin413**
Adds examples of "network" and "match" filters. 

Requires https://github.com/panther-labs/panther-enterprise/pull/7994

image

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+mypy>=0.0.0<1.0.0
+black>=22
+nose>=1.3
+panther_config>=0.0.7<=0.1.0
```
Comment (darwayne) [inline dev-requirements.txt:4]: why this version range?

Comment (darwayne) [reply dev-requirements.txt:4]: whoops was thinking panther_core nvm

Comment (nmartin413) [reply dev-requirements.txt:4]: pattern i've seen elsewhere. it's anything that's `0.0.x`

**Comment by darwayne** [comment]
what's the goal of this repo? to show example usage of panther-config? or to be a library? or both?

**Review by maxrichie5 — APPROVED** [review]
I would love to hear your reply to Dar regarding the grand plan of this. But this looks good to me.

**Review by danbiwer — APPROVED** [review]

**Comment by nmartin413** [comment]
> what's the goal of this repo? to show example usage of panther-config? or to be a library? or both?

It's an example of writing content for a panther_config based repo. You can think of this as being analogous to Pulumi's `TLS` or `random` packages.

https://www.pulumi.com/registry/packages/tls/
https://www.pulumi.com/registry/packages/random/

In the pulumi example, those packages let you create resources that aren't specific to AWS, Azure, Google Cloud, etc..

For us, `panther-utils` provides filters, etc that aren't specific to a single Log Source / Log Type (i.e. you would instead find okta specific detections in `panther-okta` - which itself may import `panther-utils`)

**Comment by maxrichie5** [comment]
> 

Is this intended to replace the current "panther_globals" functionality?

**Comment by nmartin413** [comment]
> Is this intended to replace the current "panther_globals" functionality?

Some of what this does has overlap with some of the use cases for globals today.

**Review by cdzombak — APPROVED** [review]

**Pull request merged**